doc/source/reference/arrays.datetime.rst

```diff
@@ -309,6 +309,46 @@ us / μs    microsecond      +/- 2.9e5 years         [290301 BC, 294241 AD]
    as      attosecond       +/- 9.2 seconds         [  1969 AD,   1970 AD]
 ======== ================ ======================= ==========================
 
+
+Converting datetime and timedelta to Python Object
+==================================================
+
+NumPy follows a strict protocol when converting `datetime64` and/or `timedelta64` to Python Objects (e.g., ``tuple``, ``list``, `datetime.datetime`). 
+
+The protocol is described in the following table:
+
+================================ ================================= ==================================
+         Input Type                         for `datetime64`               for `timedelta64`
+================================ ================================= ==================================
+              ``NaT``                           ``None``                           ``None``
+     Finer than Microseconds                    ``int``                            ``int``
+     Microseconds or Coarser              `datetime.datetime`              `datetime.timedelta`
+         Days or Coarser                    `datetime.date`                `datetime.timedelta`
+Non-linear(Y/M) and genric units            `datetime.date`                      ``int``
+================================ ================================= ==================================
+
+.. admonition:: Example
+
+  .. try_examples::
+
+    >>> import numpy as np
+
+    >>> type(np.datetime64('NaT').astype(object))
+    <class 'NoneType'>
+
+    >>> type(np.timedelta64('NaT').astype(object))
+    <class 'NoneType'>
+
+    >>> type(np.timedelta64(123, 'ns').astype(object))
+    <class 'int'>
+
+    >>> type(np.datetime64('2025-01-01T12:00:00.123456').astype(object))
+    <class 'datetime.datetime'>
+
+    >>> type(np.timedelta64(10, 'D').astype(object))
+    <class 'datetime.timedelta'>
```

Comment:
```suggestion
    >>> type(np.datetime64('NaT').item())
    <class 'NoneType'>

    >>> type(np.timedelta64('NaT').item())
    <class 'NoneType'>

    >>> type(np.timedelta64(123, 'ns').item())
    <class 'int'>

    >>> type(np.datetime64('2025-01-01T12:00:00.123456').item())
    <class 'datetime.datetime'>

    >>> type(np.timedelta64(10, 'D').item())
    <class 'datetime.timedelta'>
```

Besides being a bit shorter, `.item()` also has better static type annotations. In fact, it's even theoretically impossible to properly annotate `astype(object)`, since `object` is the most general type. So I guess my point is that it's best to avoid it, or at least, if you care about type checking.